dsgrid/registry/project_registry_manager.py

```diff
@@ -302,3 +358,22 @@ def remove(self, config_id):
         self._remove(config_id)
         for key in [x for x in self._projects if x.id == config_id]:
             self._projects.pop(key)
+
+    def _remap_dimension_columns(self, df, mapping_references):
+        # This will likely become a common activity when running queries.
+        # May need to cache the result. But that will likely be in Project, not here.
+        # This method could be moved elsewhere.
+        for ref in mapping_references:
+            column = ref.from_dimension_type.value
+            if column in df.columns:
+                mapping_config = self._dimension_mapping_mgr.get_by_id(
+                    ref.mapping_id, version=ref.version
+                )
+                records = models_to_dataframe(mapping_config.model.records)
+                df = (
+                    df.join(records, getattr(df, column) == records.from_id)
```

Comment:
This tells me we're dropping values in df not found in the mapping.

Reply:
My addition in https://github.com/dsgrid/dsgrid/pull/101 requires that a dimension mapping include all records, so this code is currently correct. However, I should have included an assert to make sure the length didn't change.

I actually think my earlier change is incorrect. We don't need to require that all records are covered in the mapping.